src/app/sections/payment-routing-rules/routing-rules-list/routing-rules-list.component.ts

```diff
@@ -0,0 +1,139 @@
+
+    @ViewChild(MatPaginator) set paginator(paginator: MatPaginator) {
+        this.paginator$.next(paginator);
+    }
+    private paginator$ = new ReplaySubject<MatPaginator>(1);
+
+    dataSource$ = combineLatest([
+        this.data$,
+        this.paginator$.pipe(startWith<any, null>(null)),
+    ]).pipe(
+        map(([d, paginator]) => {
+            const data = new MatTableDataSource(d);
+            data.paginator = paginator;
+            return data;
+        }),
+        shareReplay(1)
+    );
+
+    constructor(
+        private dialog: MatDialog,
+        private errorService: ErrorService,
+        private routingRulesService: RoutingRulesService
+    ) {}
+
+    getColumnsKeys(col) {
+        return col.key;
+    }
+
+    changeDelegateRuleset(delegateId: DelegateId) {
+        this.dialog
+            .open(ChangeDelegateRulesetDialogComponent, {
+                ...ChangeDelegateRulesetDialogComponent.defaultConfig,
+                data: {
+                    mainRulesetRefID: delegateId.parentRefId,
+                    delegateIdx: delegateId.delegateIdx,
+                },
+            })
+            .afterClosed()
+            .pipe(handleError(this.errorService.error), untilDestroyed(this))
+            .subscribe();
+    }
+
+    changeTarget(delegateId: DelegateId) {
+        this.dialog
+            .open(ChangeTargetDialogComponent, {
+                ...ChangeTargetDialogComponent.defaultConfig,
+                data: {
+                    mainRulesetRefID: delegateId.parentRefId,
+                    delegateIdx: delegateId.delegateIdx,
+                },
+            })
+            .afterClosed()
+            .pipe(untilDestroyed(this))
+            .subscribe({ error: this.errorService.error });
+    }
+
+    cloneDelegateRuleset(delegateId: DelegateId) {
+        this.dialog
+            .open(ConfirmActionDialogComponent)
+            .afterClosed()
+            .pipe(
+                filter((r) => r === 'confirm'),
+                switchMap(() =>
+                    this.routingRulesService.cloneDelegateRuleset({
+                        mainRulesetRefID: delegateId.parentRefId,
+                        delegateIdx: delegateId.delegateIdx,
+                    })
+                ),
+                untilDestroyed(this)
+            )
+            .subscribe({ error: this.errorService.error });
+    }
+
+    delete(delegateId: DelegateId) {
+        this.dialog
+            .open(ConfirmActionDialogComponent)
+            .afterClosed()
+            .pipe(
+                filter((r) => r === 'confirm'),
+                switchMap(() =>
+                    this.routingRulesService.deleteDelegate({
+                        mainRulesetRefID: delegateId.parentRefId,
+                        delegateIdx: delegateId.delegateIdx,
+                    })
+                ),
+                untilDestroyed(this)
+            )
+            .subscribe({ error: this.errorService.error });
```

Comment:
unreadable. It works but really unreadable

Reply:
why?